Makefile.am

```diff
@@ -187,6 +187,16 @@ check-all-install:
 trace:
 	@(cd src && $(MAKE) $(AM_MAKEFLAGS) $@) || exit 1;
 
+# Run doxygen across source files.
+# Currently, only invoke doxygen in the src directory.
+doxygen:
+	@echo Invoking doxygen
+	for d in src; do                                                \
```

Comment:
I thought you wanted to run from top directory?

Reply:
Yes, this allows you to make doxygen from the root folder, i.e. we could add c++ to the list and it would do the c++ folder in addition to the src folder.
OR
You can just run make doxygen in the sub folder.

Reply:
I know, but "d is src", src?

Reply:
yes, look at the other targets. effectively for subdir d in list[src] do {cd subdir; make doxygen;}